helpers/aws-lc/parse-functions.go

```diff
@@ -0,0 +1,105 @@
+// Adapted from the Go project's build-goboring.sh:
+//   https://github.com/golang/go/blob/master/src/crypto/internal/boring/build-goboring.sh
+
+// Original Copyright 2022 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Modifications Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var funcDefinition = regexp.MustCompile("^(const )?[^ ]+ \\**_goboringcrypto_([^(]*)\\(")
+
+var osWriteFile = os.WriteFile
+
+// main is responsible for parsing out the functions from the goboringcrypto.h file.
+// It writes it out in several formats to be used later for processing.
+func main() {
+	if len(os.Args) != 3 {
+		fmt.Println("usage: ", os.Args[0], " <path_to_goboringcrypto.h> <output_dir>")
+		os.Exit(1)
+	}
+
+	if err := writeFunctions(os.Args[1], os.Args[2]); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func writeFunctions(headerFile string, outputDir string) error {
+	syms, globals, renames, err := parse(headerFile)
+	if err != nil {
+		return err
+	}
+
+	// Write out our symbols file, ensuring it ends with a newline to prevent some unix tools from failing.
+	if err := osWriteFile(filepath.Join(outputDir, "syms.txt"), []byte(strings.Join(syms, "\n")+"\n"), 0644); err != nil {
+		return fmt.Errorf("failed writing syms.txt: %w", err)
+	}
+	// Write out our globals file, ensuring it ends with a newline to prevent some unix tools from failing.
+	if err := osWriteFile(filepath.Join(outputDir, "globals.txt"), []byte(strings.Join(globals, "\n")+"\n"), 0644); err != nil {
+		return fmt.Errorf("failed writing globals.txt: %w", err)
+	}
+	// Write out our renames file, ensuring it ends with a newline to prevent some unix tools from failing.
+	if err := osWriteFile(filepath.Join(outputDir, "renames.txt"), []byte(strings.Join(renames, "\n")+"\n"), 0644); err != nil {
+		return fmt.Errorf("failed writing renames.txt: %w", err)
+	}
+
+	return nil
+}
+
+func parse(headerFile string) ([]string, []string, []string, error) {
```

Comment:
Should we add unit tests for these functions to ensure changing in this file in future won't introduce any regression?

Reply:
based on https://github.com/bottlerocket-os/bottlerocket-sdk/pull/162#discussion_r1534725834 reply, resolving it.